evcouplings/utils/pipeline.py

```diff
@@ -472,6 +476,89 @@ def _handler(signal_, frame):
         raise
 
 
+def calculate_memory_requirements(config):
+    """
+
+    Calculates the memory requirements in MB based on the pipeline configuration
+
+    memory_in_MB = (1/2 * q^2 * (L - 1) * L  + q * L) / 12500
```

Comment:
Is this general enough for mean-field memory requirements as well?

Reply:
might not be. Have to check how Sophia implemented mean-field